### ISSUE TITLE
Non-static method

### ISSUE THREAD

**Issue by jdarwood007** [SimpleMachines/sphinx-for-smf#14]
https://example.com/index.php?action=admin;area=search
8192: Non-static method sphinxql_search::searchSettings() should not be called statically
File: /var/www/Sources/ManageSearch.php
Line: 138

**Comment by chopinesque** [comment]
I got that too


```
135: |  
-- | --
136: | $method_call = array($modSettings['search_index'] . '_search', 'searchSettings');
137: | if (is_callable($method_call))
==>138: | call_user_func_array($method_call, array(&$config_vars));
139: | }
140: |  
141: | if ($return_config)
142: | return $config_vars;


```

**Comment by qcybb** [comment]
This is a fix for SMF 2.0 :

Sources/SearchAPI-Sphinxql.php

**Find:** public function searchSettings(&$config_vars)
**Replace:** public static function searchSettings(&$config_vars)

**Find:** header('ETag: ' . sha1('sphinx.conf' + time()));
**Replace:** header('ETag: ' . sha1('sphinx.conf' . time()));
